addOns/jwt/src/main/resources/org/zaproxy/zap/extension/jwt/resources/Messages.properties

```
@@ -0,0 +1,78 @@
+jwt.scanner.client.vulnerability.cookieprefix.soln=Whenever a cookie contains sensitive information or is a session token, ensure hardening the cookie using cookie prefixes.
+jwt.scanner.client.vulnerability.cookieprefix.refs=https://googlechrome.github.io/samples/cookie-prefixes/
+
+jwt.scanner.client.vulnerability.urlparam.name=JWT token is in URL
+jwt.scanner.client.vulnerability.urlparam.desc=The request contains JWT token which is leaked in the URL. This can violate PCI and most organizational compliance policies. You can configure the list of strings for this check to add or remove values specific to your environment.
+jwt.scanner.client.vulnerability.urlparam.soln=Do not pass sensitive information in URIs.
+jwt.scanner.client.vulnerability.urlparam.refs=https://cheatsheetseries.owasp.org/cheatsheets/JSON_Web_Token_Cheat_Sheet_for_Java.html
+
+jwt.scanner.client.vulnerability.formparam.name=JWT token is in Form.
+jwt.scanner.client.vulnerability.formparam.desc=The request contains JWT token in Form param, as this is not part of cookies so assumption is that token is either present in Local Storage/Session Storage or hidden form field. In case if XSS is there then this way of storing parameter can be dangerous.
+jwt.scanner.client.vulnerability.formparam.soln=Ensure that XSS is not there in the application or Use Finger printing technique as mentioned in https://github.com/SasanLabs/JWTExtension#fingerprinting-jwt-token
+jwt.scanner.client.vulnerability.formparam.refs=https://cheatsheetseries.owasp.org/cheatsheets/JSON_Web_Token_Cheat_Sheet_for_Java.html
+
+
+# Server side JWT token vulnerabilities
+jwt.scanner.server.vulnerability.headerAttack.noneAlgorithm.name=none hashing algorithm attack
+jwt.scanner.server.vulnerability.headerAttack.noneAlgorithm.desc=JWT library accepts none hashing algorithm. none hashing algorithm is used by the JWT in case the integrity of token is already verified.so an attacker can alter the token claims and token will be trusted by the application.
+jwt.scanner.server.vulnerability.headerAttack.noneAlgorithm.soln=Not allowing none hashing algorithm.
+jwt.scanner.server.vulnerability.headerAttack.noneAlgorithm.refs=https://cheatsheetseries.owasp.org/cheatsheets/JSON_Web_Token_Cheat_Sheet_for_Java.html
+
+jwt.scanner.server.vulnerability.signatureAttack.nullByte.name=Null byte injection attack
+jwt.scanner.server.vulnerability.signatureAttack.nullByte.desc=Signature bytes after null byte are ignored ie not validated hence JWT validator is vulnerable to null byte injection
+jwt.scanner.server.vulnerability.signatureAttack.nullByte.refs=http://projects.webappsec.org/w/page/13246949/Null%20Byte%20Injection
+jwt.scanner.server.vulnerability.signatureAttack.nullByte.soln=validate entire Signature and validating library should not stop validation after null byte.
+
+jwt.scanner.server.vulnerability.signatureAttack.algorithmConfusion.name=Algorithm Confusion attack
```

```suggestion
jwt.scanner.server.vulnerability.signatureAttack.algorithmConfusion.name=Algorithm Confusion Attack
```
Attack or Vulnerability?